`yadage.manualcli.preview` failing

The CI is currently failing for PR #108 with the following error

```pytb
______________________________ test_manual_remove ______________________________

tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_manual_remove0')

    def test_manual_remove(tmpdir):
        runner = CliRunner()
        workdir = os.path.join(str(tmpdir), "workdir")
        metadir = os.path.join(str(tmpdir), "metadir")
        statefile = os.path.join(str(tmpdir), "state.json")
        result = runner.invoke(
            yadage.manualcli.init,
            [
                workdir,
                "workflow.yml",
                "-t",
                "tests/testspecs/local-helloworld",
                "-s",
                "filebacked:" + statefile,
                "-p",
                "par=value",
                "--metadir",
                metadir,
            ],
        )
        assert result.exit_code == 0
    
        result = runner.invoke(
            yadage.manualcli.preview, ["-s", "filebacked:" + statefile, "/init"]
        )
>       assert result.exit_code == 0
E       assert 1 == 0
E        +  where 1 = <Result TypeError("object of type 'dict_keyiterator' has no len()")>.exit_code
```

I'm not sure what is going on here, so if @lukasheinrich has input on this that would be great.

--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ import os
 from setuptools import setup, find_packages
 
 deps = [
-    "adage",
+    "adage>=0.10.2",  # Ensure networkx>=2.4
     "packtivity",
     "yadage-schemas",
     "click",

--- a/yadage/manualutils.py
+++ b/yadage/manualutils.py
@@ -51,7 +51,7 @@ def preview_rule(wflow, name=None, identifier=None):
         if x.identifier not in existing_rules
     ]
     new_nodes = [
-        {"name": newflow.dag.getNode(n).name, "parents": newflow.dag.predecessors(n)}
+        {"name": newflow.dag.getNode(n).name, "parents": [*newflow.dag.predecessors(n)]}
         for n in newflow.dag.nodes()
         if n not in existing_nodes
     ]

--- a/yadage/strategies.py
+++ b/yadage/strategies.py
@@ -55,7 +55,7 @@ def target(name, opts=None):
         return False, False
 
     def upstream(dag, target_identifier):
-        predecessors = dag.predecessors(target_identifier)
+        predecessors = [*dag.predecessors(target_identifier)]
         return list(
             set(predecessors + [g for p in predecessors for g in upstream(dag, p)])
         )

